es79x/src/main/scala/tech/beshu/ror/es/ReadonlyRestPlugin.scala

```diff
@@ -0,0 +1,227 @@
+import java.util.concurrent.atomic.AtomicReference
+import java.util.function.{Supplier, UnaryOperator}
+
+import monix.execution.Scheduler
+import monix.execution.schedulers.CanBlock
+import org.elasticsearch.ElasticsearchException
+import org.elasticsearch.action.support.ActionFilter
+import org.elasticsearch.action.{ActionRequest, ActionResponse}
+import org.elasticsearch.client.Client
+import org.elasticsearch.client.node.NodeClient
+import org.elasticsearch.cluster.metadata.IndexNameExpressionResolver
+import org.elasticsearch.cluster.node.DiscoveryNodes
+import org.elasticsearch.cluster.service.ClusterService
+import org.elasticsearch.common.component.LifecycleComponent
+import org.elasticsearch.common.inject.Inject
+import org.elasticsearch.common.io.stream.NamedWriteableRegistry
+import org.elasticsearch.common.network.NetworkService
+import org.elasticsearch.common.settings._
+import org.elasticsearch.common.util.concurrent.{EsExecutors, ThreadContext}
+import org.elasticsearch.common.util.{BigArrays, PageCacheRecycler}
+import org.elasticsearch.common.xcontent.NamedXContentRegistry
+import org.elasticsearch.env.{Environment, NodeEnvironment}
+import org.elasticsearch.http.HttpServerTransport
+import org.elasticsearch.index.IndexModule
+import org.elasticsearch.index.mapper.IgnoredFieldMapper
+import org.elasticsearch.indices.breaker.CircuitBreakerService
+import org.elasticsearch.plugins.ActionPlugin.ActionHandler
+import org.elasticsearch.plugins._
+import org.elasticsearch.repositories.RepositoriesService
+import org.elasticsearch.rest.{RestChannel, RestController, RestHandler, RestRequest}
+import org.elasticsearch.script.ScriptService
+import org.elasticsearch.threadpool.ThreadPool
+import org.elasticsearch.transport.{SharedGroupFactory, Transport}
+import org.elasticsearch.transport.netty4.Netty4Utils
+import org.elasticsearch.watcher.ResourceWatcherService
+import tech.beshu.ror.Constants
+import tech.beshu.ror.boot.EsInitListener
+import tech.beshu.ror.configuration.RorSsl
+import tech.beshu.ror.es.rradmin.rest.RestRRAdminAction
+import tech.beshu.ror.es.rradmin.{RRAdminActionType, TransportRRAdminAction}
+import tech.beshu.ror.es.dlsfls.RoleIndexSearcherWrapper
+import tech.beshu.ror.es.ssl.{SSLNetty4HttpServerTransport, SSLNetty4InternodeServerTransport}
+import tech.beshu.ror.utils.AccessControllerHelper.doPrivileged
+import tech.beshu.ror.es.utils.ThreadRepo
+import tech.beshu.ror.providers.{EnvVarsProvider, OsEnvVarsProvider}
+import tech.beshu.ror.buildinfo.LogPluginBuildInfoMessage
+import tech.beshu.ror.utils.SetOnce
+
+import scala.collection.JavaConverters._
+import scala.concurrent.duration._
+import scala.language.postfixOps
+
+@Inject
+class ReadonlyRestPlugin(s: Settings, p: Path)
+  extends Plugin
+    with ScriptPlugin
+    with ActionPlugin
+    with IngestPlugin
+    with NetworkPlugin
+    with ClusterPlugin {
+
+  LogPluginBuildInfoMessage()
+
+  Constants.FIELDS_ALWAYS_ALLOW.add(IgnoredFieldMapper.NAME)
```

Comment:
what about other fields?

Reply:
MapperService moved them to `META_FIELDS_BEFORE_7DOT8`, and all of them, except `IgnoredFieldMapper.NAME` are already in `FIELDS_ALWAYS_ALLOW`